code/SQLite3Query.php

```diff
@@ -39,7 +39,7 @@ public function __construct(SQLite3Connector $database, SQLite3Result $handle)
 
     public function __destruct()
     {
-        if ($this->handle) {
+        if ($this->handle && $this->database->getSelectedDatabase()) {
```

Comment:
Would `isActive()` be more appropriate to call here?

Reply:
yes is active should work just as well but brings no additional benefit except being probably more correct semantically

https://github.com/silverstripe/silverstripe-sqlite3/blob/1e8d7357be321b3eb32a596538078fd65d0d357e/code/SQLite3Connector.php#L184C21-L188

if you see, unloading the db remove the databaseName and close the connection. isActive only checks if there is a dbConn object... and there would be a closed one. :-) so either way should work just fine. i don't think there is a way to check if the dbConn was indeed closed.

Reply:
Yup that's what I was thinking too. If you change it to `isActive()` I'll be happy to merge. I like semantically correct ;p